subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -346,13 +338,12 @@ static void send_pending_adv(struct k_work *work)
 		return;
 	}
 
-	atomic_set_bit(ext_adv->flags, ADV_FLAG_PROXY_START);
-
 	if (!bt_mesh_adv_gatt_send()) {
```

Comment:
First: Please do not set `ADV_FLAG_PROXY` in advance, this will cause problem.
Second: No race condition happen here when we start connectable advertising and get connection before we return here, Since both hci command complete event&conn complete event use some thread(BT RX), fifo.

Reply:
>  Second: No race condition happen here when we start connectable advertising and get connection before we return here, Since both hci command complete event&conn complete event use some thread(BT RX), both of this are fifo process, so hci command complete event always advance process than conn complete event.

Please ignore the answer here, the latest commit has resolved this risk.